kuzzleio/kuzzle
# Add an application-level PING for WebSocket

## What does this PR do ?

We need a keep-alive system on client side.
That means that the client has to ping Kuzzle and Kuzzle response to it.
With the node library, it's automatic, the response of Kuzzle is automatic since the ping frame sent is a _real_ one.

However, through the browser that is not possible since there is no API implemented to send pings.
We needed something higher level, like a custom message to interpret.
See the related PR on the SDK JS https://github.com/kuzzleio/sdk-javascript/pull/586

:small_blue_diamond:  This tiny code came from the conclusion that's it's the only way to go after several tries. 

### How should this be manually tested?

Check out this branch of the sdk js https://github.com/kuzzleio/sdk-javascript/pull/586
Create an `index.html` and import the SDK JS
```html
<!DOCTYPE html>
<html>

<head>
  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Kuzzle SDK Playground</title>
  <script src="../sdks/sdk-javascript/dist/kuzzle.js"></script> //replace with your path
</head>

<body>
  <script>
    var kuzzle = new KuzzleSDK.Kuzzle(
      new KuzzleSDK.WebSocket('localhost')
    );
    kuzzle.on('networkError', error => {
      console.error('Network Error: ', error);
    });
    kuzzle.on('connected', () => {
      console.log('Successfully connected to Kuzzle');
    });
    kuzzle.connect();
  </script>
</body>

</html>
```
 and a `index.js` 
```js
const {
  Kuzzle,
  WebSocket
} = require('../sdks/sdk-javascript/');

const kuzzle = new Kuzzle(
  new WebSocket('localhost')
);

kuzzle.on('networkError', error => {
  console.error('Network Error:', error);
});
const run = async () => {
  try {
    // Connects to the Kuzzle server
    await kuzzle.connect();
  } catch (error) {
    console.error(error.message);
  }
};
run();
```
Add console logs after sending/receiving messages
Launch the node app and your html page
See logs and Kuzzle still alive
